src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -71,7 +72,7 @@ public class CertificateManager {
     private CertificatesConfig certificatesConfig;
     private static final Logger logger = LogManager.getLogger(CertificateManager.class);
     private static final String pkcs11Scheme = "pkcs11";
-
+    private ClientDeviceAuthMetrics metrics;
```

Comment:
You're never setting the value of this object.

Inject the value in the constructor on L90.